File: app/src/main/java/com/artemchep/horario/_new/adapters/UsersAdapter.java
package com.artemchep.horario._new.adapters;

import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.artemchep.horario.R;
import com.artemchep.horario._new.fragments.FragmentModel;
import com.artemchep.horario.database.models.User;
import com.artemchep.horario.ui.widgets.UserView;

import java.util.List;

/**
 * @author Artem Chepurnoy
 */
public class UsersAdapter extends AdapterModel<User, UsersAdapter.ViewHolder> {

    public UsersAdapter(@NonNull FragmentModel<User> fragment, @NonNull List<User> list) {
        super(fragment, list);
    }

    /**
     * @author Artem Chepurnoy
     */
    public static class ViewHolder extends AdapterModel.ViewHolder<User> {

        @NonNull
        final UserView userView;

        public ViewHolder(View itemView, FragmentModel<User> fragment) {
            super(itemView, fragment);
            userView = itemView.findViewById(R.id.user);
        }
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View v = inflater.inflate(R.layout.___item_user, parent, false);
        return new ViewHolder(v, mFragment);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);
        User user = getItem(position);
        holder.userView.bind(user);
    }

}

File: app/src/main/java/com/artemchep/horario/ui/adapters/UsersAdapter.kt
package com.artemchep.horario.ui.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.artemchep.horario.R
import com.artemchep.horario.database.models.IUser
import com.artemchep.horario.ui.widgets.UserView

/**
 * @author Artem Chepurnoy
 */
open class UsersAdapter<out T : IUser>(list: List<T>) : AdapterBase<T, UsersAdapter<T>.ViewHolder>(list) {

    var yourUserId: String? = null

    /**
     * @author Artem Chepurnoy
     */
    inner open class ViewHolder(v: View) : RecyclerView.ViewHolder(v), View.OnClickListener {

        internal val userView: UserView = v.findViewById(R.id.user)

        init {
            v.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            val listener = onItemClickListener
            if (listener != null) {
                val pos = adapterPosition
                if (pos != -1) listener.onItemClick(view, pos)
            }
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val v = inflater.inflate(R.layout.___item_user, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(h: ViewHolder, position: Int) {
        val user = getItem(position)
        h.userView.model = user
    }

}
